### PR TITLE
Default port

### PR DESCRIPTION
**Pull request by sidney-katz**
The file is location under LocationBuilder

### CODE REVIEW

File: data_io/connection.py [hunk 1]
```diff
@@ -732,6 +745,10 @@ class OracleConnection(DatabaseConnection):
     def jdbc_url(self) -> str:
 
         location = self.location
+        # if(location.port == None):
```
Comment (chris-bishop-tfs) [inline data_io/connection.py:748]: Remove these.

File: data_io/__init__.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
-from . import location
-from . import connection
-from . import data_source
+# from . import location
```
Comment (chris-bishop-tfs) [inline data_io/__init__.py:1]: Remove

File: data_io/connection.py [hunk 3]
```diff
@@ -37,6 +38,17 @@ def location(self):
 
     @property
     def jdbc_url(self):
+
+        # setting default for port for redshift and oracle
```
Comment (chris-bishop-tfs) [inline data_io/connection.py:42]: Remove

File: data_io/location.py [hunk 4]
```diff
@@ -10,10 +10,11 @@ class Location(URL):
     stub for extension
     """
 
-    def __init__(self, *largs, **kwargs):
+    def __init__(self, url, *largs, **kwargs):
```
Comment (chris-bishop-tfs) [inline data_io/location.py:13]: I don't understand this change

File: data_io/location.py [hunk 5]
```diff
@@ -25,7 +26,19 @@ def __init__(self, *largs, **kwargs):
         super(URLKeyBuilder, self).__init__()
 
     def build(self, url: str) -> Location:
-
+        '''
+            default ports in builder
+        '''
+        path = URL(url)
+        if((path.port == None) and (path.scheme != 's3a')):
```
Comment (chris-bishop-tfs) [inline data_io/location.py:33]: Dictionary would be more ergonomic here

**Review by chris-bishop-tfs — CHANGES_REQUESTED** [review]

File: data_io/location.py (hunk 4 above)
Comment (sidney-katz) [reply data_io/location.py:13]: Initially was going to have default port's in location

**Comment by sidney-katz** [comment]
Uploaded default ports into dictionary

File: data_io/__init__.py (hunk 2 above)
Comment (chris-bishop-tfs) [reply data_io/__init__.py:1]: Remove

File: data_io/__init__.py [hunk 6]
```diff
@@ -1,8 +1,11 @@
...
+# from . import connection
+# from . import data_source
+# from . import get_credentials
+# from . import test_connection
```
Comment (chris-bishop-tfs) [inline data_io/__init__.py:5]: Remove

File: data_io/builder.py [hunk 7]
```diff
@@ -82,10 +82,14 @@ def build_url_key(self, url: str) -> str:
         # This will be parsed, etc. correctly
         location = URL(url)
 
+
```
Comment (chris-bishop-tfs) [inline data_io/builder.py:85]: Remove extra spacing

File: data_io/builder.py [hunk 8]
```diff
@@ -82,10 +82,14 @@ def build_url_key(self, url: str) -> str:
...
+
+
         # Build the connection key from the URL
         #  scheme (protocol), extension?
         #
         #  KISS and use protocol only to start
         url_key = (location.scheme,)
+        # url_key = location
```
Comment (chris-bishop-tfs) [inline data_io/builder.py:93]: Remove testing lines/edits

File: data_io/location.py [hunk 9]
```diff
@@ -25,8 +26,18 @@ def __init__(self, *largs, **kwargs):
         super(URLKeyBuilder, self).__init__()
 
     def build(self, url: str) -> Location:
-
-        # Build the URL key
+        '''
+            default ports in builder
+        '''
+        # dictionary of defaul port
+        default_port = dict(oracle = '1521', redshift = '5439')
```
Comment (chris-bishop-tfs) [inline data_io/location.py:33]: Recommend changing the dictionary keys to the connection lookup keys (tuples).

Also, keyword args do not have spaces before/after assignment operator.

Example:

`oracle='1521'`

**Review by chris-bishop-tfs — CHANGES_REQUESTED** [review]

File: data_io/connection.py [hunk 10]
```diff
@@ -37,6 +38,9 @@ def location(self):
 
     @property
     def jdbc_url(self):
+
```
Comment (chris-bishop-tfs) [inline data_io/connection.py:41]: remove

**Review by chris-bishop-tfs — CHANGES_REQUESTED** [review]

File: data_io/location.py [hunk 11]
```diff
@@ -13,7 +13,8 @@ class Location(URL):
     def __init__(self, *largs, **kwargs):
         super(URL, self).__init__()
```
Comment (bryan-yekelchik) [inline data_io/location.py:15]: extra space here to remove

**Review by bryan-yekelchik — CHANGES_REQUESTED** [review]

**Pull request merged**